keyboards/skyloong/gk87/ptir/ptir.c

```diff
@@ -0,0 +1,264 @@
+    {1, CS4_SW9,   CS5_SW9,   CS6_SW9},
+    {1, CS4_SW10,  CS5_SW10,  CS6_SW10},
+    {1, CS4_SW11,  CS5_SW11,  CS6_SW11},
+    {1, CS13_SW2,  CS14_SW2,  CS15_SW2},
+    {1, CS13_SW4,  CS14_SW4,  CS15_SW4},
+
+
+    {1, CS7_SW1,   CS8_SW1,   CS9_SW1},
+    {1, CS7_SW2,   CS8_SW2,   CS9_SW2},
+    {1, CS7_SW3,   CS8_SW3,   CS9_SW3},
+    {1, CS7_SW5,   CS8_SW5,   CS9_SW5},
+    {1, CS7_SW6,   CS8_SW6,   CS9_SW6},
+    {1, CS7_SW8,   CS8_SW8,   CS9_SW8},
+    {1, CS7_SW10,  CS8_SW10,  CS9_SW10},
+    {1, CS7_SW11,  CS8_SW11,  CS9_SW11},
+    {1, CS16_SW1,  CS17_SW1,  CS18_SW1},
+    {1, CS16_SW2,  CS17_SW2,  CS18_SW2},
+    {1, CS16_SW3,  CS17_SW3,  CS18_SW3},
+    {1, CS16_SW4,  CS17_SW4,  CS18_SW4},
+    {1, CS16_SW5,  CS17_SW5,  CS18_SW5}
+};
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (!process_record_user(keycode, record)) {
+        return false;
+    }
+
+    switch (keycode) {
+#    ifdef RGB_MATRIX_ENABLE
+        case RGB_TOG:
+            if (record->event.pressed) {
+                switch (rgb_matrix_get_flags()) {
+                    case LED_FLAG_ALL: {
+                        rgb_matrix_set_flags(LED_FLAG_NONE);
+                        rgb_matrix_set_color_all(0, 0, 0);
+                    } break;
+                    default: {
+                        rgb_matrix_set_flags(LED_FLAG_ALL);
+                    } break;
+                }
+            }
+         return false;
+#    endif
+     case TO(0):
+      if (record->event.pressed) {
+       L_WIN = 1;
+      } else {
+       L_WIN = 0;
+      }
+      return true; // continue all further processing of this key
+
+     case MO(2):
+      if (record->event.pressed) {
+       FN_WIN = 1;
+      } else {
+       FN_WIN = 0;
+      }
+      return true; // continue all further processing of this key
+
+     case TO(1):
+      if (record->event.pressed) {
+       L_MAC = 1;
+      } else {
+       L_MAC = 0;
+      }
+      return true; // continue all further processing of this key
+
+     case MO(3):
+      if (record->event.pressed) {
+       FN_MAC = 1;
+      } else {
+       FN_MAC = 0;
+      }
+      return true; // continue all further processing of this key
+    default:
+      return true;
+    }
+}
+
+bool rgb_matrix_indicators_advanced_kb(uint8_t led_min, uint8_t led_max) {
+    if (!rgb_matrix_indicators_advanced_user(led_min, led_max)) {
+        return false;
+    }
+
+   switch (get_highest_layer(layer_state)) {
+      case 2:{
+        RGB_MATRIX_INDICATOR_SET_COLOR(WIN_MOD_INDEX, 255, 255, 255);
+        if (!rgb_matrix_get_flags()) {
+            RGB_MATRIX_INDICATOR_SET_COLOR(MAC_MOD_INDEX, 0, 0, 0);
+         }
+      } break;
+      case 3:{
+        RGB_MATRIX_INDICATOR_SET_COLOR(MAC_MOD_INDEX, 255, 255, 255);
+        if (!rgb_matrix_get_flags()) {
+            RGB_MATRIX_INDICATOR_SET_COLOR(WIN_MOD_INDEX, 0, 0, 0);
+         }
+      } break;
+
+      case 0:{
+       if (L_WIN) {
```

Comment:
```suggestion
       if (!layer_state_is(1) {
```

This checks to see if layer 1 is on, rather than checking a locally set boolean value. (note that 0 is always on, so you need to check layer 1+)

Reply:
It is to check if it is TO(0) keycode and mark it.